packages/fxa-settings/src/lib/context/model-validation.ts

```diff
@@ -32,6 +32,8 @@ export class ContextValidationErrors extends Error {
   }
 }
 
+const emailRegex = /^[\w.!#$%&'*+\/=?^`{|}~-]{1,64}@[a-z\d](?:[a-z\d-]{0,253}[a-z\d])?(?:\.[a-z\d](?:[a-z\d-]{0,253}[a-z\d])?)+$/i;
```

Comment:
nit: This looks like a copy/pasta of [`packages/fxa-shared/email/helpers.ts:35`](https://github.com/mozilla/fxa/blob/790de60896631f9e2ba5c5ee30cac8ea84bce6d0/packages/fxa-shared/email/helpers.ts#L22-L35). Not sure if we should try exporting/importing that to keep a single source of RegExp chaos.

Reply:
Ah, I thought I got it from content server somewhere, but since it's in shared this seems like a good call 👍 . Nothing worse than two slightly different regexs 😬 .

Reply:
These were the only two instances of `{0,253}` I could find [in `main`]:

```sh
git grep -En "\{0,253\}" packages | cat

packages/fxa-auth-server/lib/routes/validators.js:169:  /^[A-Z0-9](?:[A-Z0-9-]{0,253}[A-Z0-9])?(?:\.[A-Z0-9](?:[A-Z0-9-]{0,253}[A-Z0-9])?)+$/i;
packages/fxa-shared/email/helpers.ts:35:const emailRegex = /^[\w.!#$%&'*+\/=?^`{|}~-]{1,64}@[a-z\d](?:[a-z\d-]{0,253}[a-z\d])?(?:\.[a-z\d](?:[a-z\d-]{0,253}[a-z\d])?)+$/i;
```

So yeah, it looks like fxa-auth-server _might_ be using a slightly different RegExp validator. Also in that same file:
https://github.com/mozilla/fxa/blob/92f6ddb07295bbd4b6ba98d99835bf675e4f4abf/packages/fxa-auth-server/lib/routes/validators.js#L171-L190

Which is a bit weird, because "back in the day" we had these bugs a few times where emails were valid on the front end but then failed in the back end because we had different rules/RegExps. :shrug: